### ISSUE TITLE
Search panel bug

### ISSUE THREAD

**Issue by MinhNguyenVan7** [microsoft/vscode#152594]
We have written the needed data into your clipboard because it was too 
Issue Type: <b>Bug</b>

In search panel, when we enable "Match Whole Word" button, with Unicode text having accentuated language (vietnamese for instance), the match is perturbed and erroneous. Presumably it only runs well with ASCII text.

VS Code version: Code 1.68.1 (30d9c6cd9483b2cc586687151bcbcd635f373630, 2022-06-14T12:48:58.283Z)
OS version: Windows_NT x64 10.0.19044
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7660U CPU @ 2.50GHz (4 x 2496)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.92GB (8.53GB free)|
|Process Argv|--crash-reporter-id f24437c4-2918-402a-b294-f879f28257c7 --crash-reporter-id f24437c4-2918-402a-b294-f879f28257c7|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (1)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-language-pack-fr|MS-|1.68.6150916

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscoreces:30445986
pythondataviewer:30285071
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
vsc1dst:30438360
pythonvs932:30410667
wslgetstarted:30449410
vscscmwlcmt:30465135
cppdebug:30492333
vsclangdc:30486549

```

</details>

large to send. Please paste.

**Comment by roblourens** [comment]
Looks ok to me, can you share a specific example? 

<img width="529" alt="image" src="https://user-images.githubusercontent.com/323878/174877044-29c9ab35-1136-4aa2-8bfa-b3b045cef6b9.png">

**Comment by MinhNguyenVan7** [comment]
Thank you of your fast response.

First of all, let me say VSCode (and VSCodium) is the far best multi-file editor for me (I compared with at least 6 other well known competitors) !

In seeing your test, I did tests on very short .txt then .epub files. Everything is then OK.
This bug only appears in long .epub files I use (presently 8 Mo).
I sent to you a screen copy with 2 underlines to show the bug. I have other examples.

![Vi epub test](https://user-images.githubusercontent.com/107808094/174971577-49a467db-8cce-4792-a530-77e58172ec2c.png)
